src/Grpc.AspNetCore.Server/Internal/ServerStreamingServerCallHandler.cs

```diff
@@ -55,12 +55,19 @@ public override async Task HandleCallAsync(HttpContext httpContext)
             // Activate the implementation type via DI.
             var activator = httpContext.RequestServices.GetRequiredService<IGrpcServiceActivator<TService>>();
             var service = activator.Create();
+            var serverCallContext = new HttpContextServerCallContext(httpContext);
+            var streamWriter = new HttpContextStreamWriter<TResponse>(serverCallContext, Method.ResponseMarshaller.Serializer);
 
-            await _invoker(service, request, new HttpContextStreamWriter<TResponse>(httpContext, Method.ResponseMarshaller.Serializer), null);
+            await _invoker(
```

Comment:
one thing that's missing here is to handle RpcException thrown by user's code.
That's an alternative way to tell the handling stack to set the status and trailers that are set in the exception.

See https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core/Internal/ServerCallHandler.cs#L296.

Same of other call types.

I think it's fine to address in a followup PR.

Reply:
Yes I was thinking that work belongs in https://github.com/grpc/grpc-dotnet/issues/47.